layout-server/src/main/java/org/finos/vuu/layoutserver/exceptions/GlobalExceptionHandler.java

```diff
@@ -1,40 +1,53 @@
 package org.finos.vuu.layoutserver.exceptions;
 
-import java.util.List;
-import java.util.NoSuchElementException;
-import java.util.stream.Collectors;
+import org.finos.vuu.layoutserver.dto.response.ErrorResponse;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
 
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
 @ControllerAdvice
 public class GlobalExceptionHandler {
 
     @ExceptionHandler(NoSuchElementException.class)
-    public ResponseEntity<String> handleNotFound(NoSuchElementException ex) {
-        return new ResponseEntity<>(ex.getMessage(),
-            org.springframework.http.HttpStatus.NOT_FOUND);
+    public ResponseEntity<Object> handleNotFound(HttpServletRequest request, Exception ex) {
+        return generateResponse(request, ex, HttpStatus.NOT_FOUND);
     }
 
     @ExceptionHandler({
-        HttpMessageNotReadableException.class,
-        MethodArgumentTypeMismatchException.class})
-    public ResponseEntity<String> handleBadRequest(Exception ex) {
-        return new ResponseEntity<>(ex.getMessage(),
-            org.springframework.http.HttpStatus.BAD_REQUEST);
+            HttpMessageNotReadableException.class,
+            MethodArgumentTypeMismatchException.class})
+    public ResponseEntity<Object> handleBadRequest(HttpServletRequest request, Exception ex) {
+
+        return generateResponse(request, ex, HttpStatus.BAD_REQUEST);
     }
 
     @ExceptionHandler(MethodArgumentNotValidException.class)
     public ResponseEntity<String> handleMethodArgumentNotValid(MethodArgumentNotValidException ex) {
         List<String> errors = ex.getFieldErrors()
-            .stream()
-            .map(fieldError -> fieldError.getField() + ": " + fieldError.getDefaultMessage())
-            .collect(Collectors.toList());
-
+                .stream()
+                .map(fieldError -> fieldError.getField() + ": " + fieldError.getDefaultMessage())
+                .collect(Collectors.toList());
         return new ResponseEntity<>(errors.toString(),
-            org.springframework.http.HttpStatus.BAD_REQUEST);
+                org.springframework.http.HttpStatus.BAD_REQUEST);
+    }
+    @ExceptionHandler(InternalServerErrorException.class)
+    public ResponseEntity<Object> handleInternalServerError(HttpServletRequest request, Exception ex) {
+        return generateResponse(request, ex, HttpStatus.INTERNAL_SERVER_ERROR);
     }
+
+    private ResponseEntity<Object> generateResponse(HttpServletRequest request,
```

Comment:
Lets change this return type to `ResponseEntity<ErrorResponse>`, and propagate to all exception handler methods too